File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = set()
        winSum, maxSum = 0, 0
        wStart = 0
        
        for wEnd in range(len(nums)):
            while nums[wEnd] in seen:
                seen.remove(nums[wStart])
                winSum -= nums[wStart]
                wStart += 1
            
            winSum += nums[wEnd]
            seen.add(nums[wEnd])
            maxSum = max(maxSum, winSum)
        return maxSum

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        return daily(nums);
    }
private:
    int daily(vector<int>& nums) {
        set<int> seen;
        int wStart = 0;
        int maxSum = INT_MIN, winSum = 0;
        
        for(int wEnd = 0; wEnd < nums.size(); ++wEnd) {
            while(seen.find(nums[wEnd]) != seen.end()) {
                seen.erase(nums[wStart]);
                winSum -= nums[wStart++];
            }
            winSum += nums[wEnd];
            maxSum = max(maxSum, winSum);
            seen.insert(nums[wEnd]);
        }
        return maxSum;
    }
};